### PR TITLE
feat(BUILD-11337): add ruby toolchain

### PR DESCRIPTION
**Pull request by zakhark**
This adds ruby toolchain module for ubuntu 16.04
First commit was the cookiecutter template.

### CODE REVIEW

File: molecule/default/converge.yml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+---
+- become: true
+  name: Converge
+  hosts: all
+  roles:
+    - role: ansible-role-ruby-toolchain
+      vars:
+        #ruby_toolchain_url: "https://s3.amazonaws.com/boxes.10gen.com/build/toolchain-drivers/mongo-ruby-driver/ruby-toolchain-ubuntu1604-a650fe6064a65922754444f3fe04c248cdf50ef1.tar.gz"
```
Comment (jasonwalsh) [inline molecule/default/converge.yml:8]: Remove this line

File: molecule/default/converge.yml [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+        ruby_toolchain_sha: "a650fe6064a65922754444f3fe04c248cdf50ef1"
+        homedir: "/root"
```
Comment (jasonwalsh) [inline molecule/default/converge.yml:10]: Let's not introduce this variable. Instead, we should use `{{ ansible_env.HOME }}`.

File: tasks/main.yml [hunk 3]
```diff
@@ -1 +1,28 @@
...
+- name: Create version file [{{ ruby_toolchain_sha }}]
+  become: true
+  copy:
+    content: "{{ ruby_toolchain_sha }}"
+    dest: "{{ ruby_toolchain_final_dest }}/toolchain_version"
+
+- name: Link rubies to homedir
+  file:
+    src: "{{ ruby_toolchain_final_dest }}"
+    dest: "{{ homedir }}/.rubies"
```
Comment (jasonwalsh) [inline tasks/main.yml:27]:
```yaml
dest: "{{ ansible_env.HOME }}/.rubies"
```

Comment (zakhark) [reply tasks/main.yml]: Done

File: molecule/default/converge.yml (hunk 2 above)
Comment (zakhark) [reply molecule/default/converge.yml]: Done

File: molecule/default/converge.yml (hunk 1 above)
Comment (zakhark) [reply molecule/default/converge.yml]: Done

**Pull request merged**